javatests/arcs/core/analysis/testdata/fail_type_variables.arcs

```diff
@@ -0,0 +1,51 @@
+// #Ingress: Mixed
+// #Ingress: Meh
+// #FAIL: hc:FailingCheck.input.bad is bad
+particle Mixed
+  output: writes Mixed {good: Text, bad: Text}
+  claim output.good is good
+  claim output.bad is bad
+
+particle Meh
+  output: writes Meh {meh: Text}
+  claim output.meh is meh
+
+particle DoStuff
+  input1: reads ~a with {bad: Text}
+  input2: reads ~b with {*}
```

Comment:
On another though, we could also add a read with a type variable without any constraint, which should not influence the data flow and therefore validate that if there are no listed fields, then no data is accessed.

E.g. We could add this:
```
particle Ambivalent
  output: write Ambivalent {ambivalent: Text}
  claim output.ambivalent is ambivalent

particle DoStuff
  (...)
  input3: reads ~c
  (...)

recipe
  Ambivalent
    output: ambivalent
  (...)
  DoStuff
    (...)
    input3: ambivalent
  (...)
```

And maybe fold PassingCheck1  and PassingCheck2 into this new particle with an extra check: 
```
particle PassingChecks
  input: reads Mixed {good: Text, bad: Text}
  check input.good is good
  check input.bad is bad or is meh
  check input is not ambivalent
```

WDYT?

Reply:
As we chatted offline, `input3: reads ~c` is treated as `input3: reads ~c with {*}` until that information is plumbed through.  I have updated the PR just for the record.

I will merge the `PassingCheck1` and `PassingCheck2` particles into one.